codegen/src/types/mod.rs

```diff
@@ -302,12 +303,13 @@ impl<'a> TypeGenerator<'a> {
     }
 
     /// Returns the derives to be applied to a generated type.
-    pub fn type_derives(&self, ty: &Type<PortableForm>) -> Derives {
+    pub fn type_derives(&self, ty: &Type<PortableForm>) -> Result<Derives, CodegenError> {
         let joined_path = ty.path().segments().join("::");
-        let ty_path: syn::TypePath = syn::parse_str(&joined_path).unwrap_or_else(|e| {
-            abort_call_site!("'{}' is an invalid type path: {:?}", joined_path, e,)
-        });
-        self.derives.resolve(&ty_path)
+        let ty_path: syn::TypePath = match syn::parse_str(&joined_path) {
+            Ok(ty_path) => ty_path,
+            Err(e) => return Err(CodegenError::InvalidTypePath(joined_path, e)),
+        };
```

Comment:
I wonder whether the `let else` stuff can be used (untested!):

```suggestion
        let Ok(ty_path) = syn::parse_str::<syn::TypePath>(&joined_path) else {
            return Err(CodegenError::InvalidTypePath(joined_path, e));
        };
```

Reply:
`map_err` did the trick for this one, thanks!